### PR TITLE
[TimeAxis] Fixes inconsitent end tick issues

### LINKED ISSUE #1509: Inconsistent end ticks on left vs. right of time axis
screen shot 2015-01-09 at 3 21 15 pm

This is the stocks quicktest. The long tick on the left side is always there when panning, but the right side never is. This looks broken.

### PR DESCRIPTION
**Pull request by rcchen**
Fixes #1509

### CODE REVIEW

File: src/components/axes/timeAxis.ts [hunk 1]
```diff
@@ -348,10 +348,10 @@ export module Axis {
       var anchor = (this._tierLabelPositions[index] === "center" || config.step === 1) ? "middle" : "start";
       tickLabels.selectAll("text").text(config.formatter).style("text-anchor", anchor);
       if (filteredTicks.indexOf(this._scale.domain()[0]) === -1) {
-        filteredTicks.splice(0, 0, this._scale.domain()[0]);
+        filteredTicks.splice(0, 1, this._scale.domain()[0]);
```
Comment (jtlan) [inline src/components/axes/timeAxis.ts]: MFW bug caused by different API between `.splice()` and `.slice()`.

**Comment by jtlan** [comment]
Looks good! Write a small test to confirm that the ticks are drawn in the right spot.

File: test/components/timeAxisTests.ts [hunk 2]
```diff
@@ -105,4 +105,18 @@ describe("TimeAxis", () => {
...
+    var width = 500;
+    var svg = generateSVG(width, 100);
+    scale.domain(["2010", "2014"]);
+    axis.renderTo(svg);
+    var firstTick = d3.select(".tick-mark");
+    assert.equal(0, firstTick.attr("x1"));
+    assert.equal(0, firstTick.attr("x2"));
+    var lastTick = d3.select(d3.selectAll(".tick-mark")[0].pop());
+    assert.equal(width, lastTick.attr("x1"));
+    assert.equal(width, lastTick.attr("x2"));
```
Comment (jtlan) [inline test/components/timeAxisTests.ts]: [minor]: Please add a message to all these `assert` statements -- it helps identify what the test does, particularly when the assert fails.

File: test/tests.js [hunk 3]
```diff
@@ -562,6 +562,19 @@ describe("TimeAxis", function () {
...
         svg.remove();
     });
+    it("renders end ticks on either side", function () {
+        var width = 500;
+        var svg = generateSVG(width, 100);
+        scale.domain(["2010", "2014"]);
+        axis.renderTo(svg);
+        var firstTick = d3.select(".tick-mark");
+        assert.equal(0, firstTick.attr("x1"));
+        assert.equal(0, firstTick.attr("x2"));
```
Comment (rcchen) [inline test/tests.js]: Huh that's strange...

File: test/components/timeAxisTests.ts [hunk 4]
```diff
@@ -105,4 +105,18 @@ describe("TimeAxis", () => {
...
+    var width = 500;
+    var svg = generateSVG(width, 100);
+    scale.domain(["2010", "2014"]);
+    axis.renderTo(svg);
+    var firstTick = d3.select(".tick-mark");
+    assert.equal(0, firstTick.attr("x1"), "xPos (x1) of first end tick is not at the beginning of the axis container");
+    assert.equal(0, firstTick.attr("x2"), "xPos (x2) of first end tick is not at the beginning of the axis container");
+    var lastTick = d3.select(d3.selectAll(".tick-mark")[0].pop());
+    assert.equal(width, lastTick.attr("x1"), "xPos (x1) of last end tick is not at the end of the axis container");
+    assert.equal(width, lastTick.attr("x2"), "xPos (x2) of last end tick is not at the end of the axis container");
```
Comment (jtlan) [inline test/components/timeAxisTests.ts]: "is not"? Usually the style is
assert.method(*actual*, *expected*, *message on success*)

**Comment by crmorford** [comment]
this looks good!

**Pull request merged**